From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 15 May 2026 00:00:00 -0400
Subject: [PATCH] Add absl/strings:charset target to system absl BUILD

Vendored protobuf 6.31.1 references @com_google_absl//absl/strings:
charset, which the systemlib BUILD did not declare. charset is
header-only in abseil.
---
diff --git a/third_party/xla/third_party/absl/system.absl.strings.BUILD b/third_party/xla/third_party/absl/system.absl.strings.BUILD
index c8196e5c..00000000 100644
--- a/third_party/xla/third_party/absl/system.absl.strings.BUILD
+++ b/third_party/xla/third_party/absl/system.absl.strings.BUILD
@@ -55,3 +55,11 @@ cc_library(
         "//absl/base:nullability",
     ],
 )
+
+cc_library(
+    name = "charset",
+    deps = [
+        ":string_view",
+        "//absl/base:config",
+    ],
+)
